### PR TITLE
[IN PROGRESS] Combine all tests into suites

### PR DESCRIPTION
**Pull request by farodin91**
**This PR is in progress. I would like to get some feedback.** 

During my working on testcontainers, I had to write multiple times the same code because for each backend test class i need to update the configuration and i saw the solution from gremlin with test suite and started adopt this solution for janusgraph. 
This solution allows to write easily support for a new backend by adding two files, a test suite runner and a backend conection provider.

**TODO**
- hadoop backend
- solr backend
- es backend
- cassandra backend
- hbase backend
- cql backend
- support to run single tests against backends during development

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by farodin91** [comment]
I will add CLA later. If it is okay

**Comment by farodin91** [comment]
@pluradj Did all tinkerpop tests passes for in-memory?

File: janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyGraphTinkerpopTest.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.janusgraph;
```
Comment (mbrukman) [inline janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyGraphTinkerpopTest.java]: Please add the standard copyright header at the top of this and other new files.

File: janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyJanusGraphDatabaseProvider.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package org.janusgraph;
```
Comment (mbrukman) [inline janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyJanusGraphDatabaseProvider.java]: Please add the standard copyright header at the top of this and other new files.

File: janusgraph-test/src/main/java/org/janusgraph/JanusGraphDatabaseProvider.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package org.janusgraph;
```
Comment (mbrukman) [inline janusgraph-test/src/main/java/org/janusgraph/JanusGraphDatabaseProvider.java]: Please add the standard copyright header at the top of this and other new files.

File: janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryGraphBackendTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+package org.janusgraph.inmemory;
```
Comment (mbrukman) [inline janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryGraphBackendTest.java]: Please add the standard copyright header at the top of this and other new files.

File: janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryJanusGraphDatabaseProvider.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
+package org.janusgraph.inmemory;
```
Comment (mbrukman) [inline janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryJanusGraphDatabaseProvider.java]: Please add the standard copyright header at the top of this and other new files.

**Comment by farodin91** [comment]
@mbrukman I did all the changes, but i would like to get some review for it. This Pull request is more like a proof of concept.

File: janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryGraphBackendTest.java (hunk 4 above)
Comment (mbrukman) [reply janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryGraphBackendTest.java]: I see this was done.

File: janusgraph-test/src/main/java/org/janusgraph/JanusGraphDatabaseProvider.java (hunk 3 above)
Comment (mbrukman) [reply janusgraph-test/src/main/java/org/janusgraph/JanusGraphDatabaseProvider.java]: I see this was done.

File: janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryJanusGraphDatabaseProvider.java (hunk 5 above)
Comment (mbrukman) [reply janusgraph-test/src/test/java/org/janusgraph/inmemory/InMemoryJanusGraphDatabaseProvider.java]: I see this was done.

File: janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyGraphTinkerpopTest.java (hunk 1 above)
Comment (mbrukman) [reply janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyGraphTinkerpopTest.java]: I see this was done.

File: janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyJanusGraphDatabaseProvider.java (hunk 2 above)
Comment (mbrukman) [reply janusgraph-berkeleyje/src/test/java/org/janusgraph/BerkeleyJanusGraphDatabaseProvider.java]: I see this was done.

**Comment by farodin91** [comment]
I will close this for now in order to my new PR #1327.

**Pull request closed without merging**